src/utils.js

```diff
@@ -157,17 +157,50 @@ export function parseGPTSingleSizeArray(singleSize) {
   }
 };
 
-exports.getTopWindowLocation = function () {
-  let location;
+export function getTopWindowLocation() {
+  if (exports.inIframe()) {
+    return getIframeParentLoc();
+  } else {
+    return window.location;
+  }
+}
+
+const getIframeParentLoc = function() {
+  let loc = window.location;
   try {
-    // force an exception in x-domain enviornments. #1509
-    window.top.location.toString();
```

Comment:
Given the comment, not sure removing this is the correct thing to do, unless there's an equivalent fix in the new code i'm not seeing?

Reply:
Yeah, I didn't realize that the exception wasn't thrown when trying to access window.top until I read that 1509 PR. Interesting. Thanks for pointing out. I added back in.